### ISSUE TITLE
Option to remove menu bar extra (icon) on Mac OS, with e

### ISSUE THREAD

**Issue by waded** [docker/roadmap#281]
**Tell us about your request**
As a user of Docker Desktop, who runs it on startup, but likes to keep the menu bar area clear - an icon telling me Docker's running isn't useful to me, because it's supposed to be running - I'd like that icon to be a *preference* rather than required.

I'd expect the preference to be accessible by running /Applications/Docker Desktop.app or similar - to be in the same window as "Start Docker (Desktop) when you log in" for example.

Any options specific to the icon's menu (notably, and possibly only, the option that stops the service) should be available within the Docker Desktop window/app. I would expect that window/app can be quit, its Dock icon closed, without stopping the service.

The icon/menu that should be optional is pictured here:

![image](https://user-images.githubusercontent.com/1535656/140622106-a91a0f99-90ab-4f93-8f23-55130969dbba.png)

**Which service(s) is this request for?**
Docker Desktop (for Mac)

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Apple HIG suggests "menu bar extras" should be optional ("Users, not apps, should choose when a menu bar extra is added to the menu bar.  Typically, this is done by changing a setting in an app’s preferences window.") You'll find a lot of Mac users are particular about what's shown there, and rely on developers following this HIG.

**Are you currently working around the issue?**
No known workaround with Docker Desktop. The icon cannot be removed e.g. by command-click and drag - it can only be rearranged with respect to the other icons.

**Comment by martinemde** [comment]
This is especially important on a notched 14" MacBook Pro which has a very constrained menu bar space.

**Comment by juliekopackova** [comment]
Yes! I don't really understand how any application can be so arrogant and not to have this option even though it's in the MacOS design guidelines
https://developer.apple.com/design/human-interface-guidelines/macos/extensions/menu-bar-extras/

**Comment by landsman** [comment]
pls!

<img width="412" alt="Screenshot 2024-01-04 at 20 36 20" src="https://github.com/docker/roadmap/assets/3184228/24f59b0a-1dfc-4492-b982-8e0c9ab139d1">

**Comment by tovbinm** [comment]
Yes, please add!

**Comment by klippx** [comment]
How hard can it be? People have been asking for this for several years. I agree with @julierakovcova that the right word is arrogance, but by now its downright negligence.

**Comment by martinemde** [comment]
I just use OrbStack now instead of Docker. I highly recommend it.

**Comment by antonsacred** [comment]
I vote for this bug/feature to be solved

**Comment by palaashatri** [comment]
Hi
Is there any progress on this?

Pretty sure its a high priority feature request from multiple users!

**Comment by Javernus** [comment]
It's been three years, they don't seem to care. 

What I do now that could circumvent this problem is [Ice](https://github.com/jordanbaird/Ice), which is an app that lets you customise your menu bar (i.e. hide icons). It's totally free and can be installed with brew (`brew install jordanbaird-ice`).

**Comment by davidgfitzgerald** [comment]
Over 3 years! This must be easy to implement!

**Comment by ltrnvn** [comment]
Still waiting for this in 2025...

**Comment by jakeleventhal** [comment]
MacOS 26 implemented the ability to remove what you want. Should close this issue

**Comment by Javernus** [comment]
It is great that Apple finally stood up to applications hoarding menu space against their human interface guidelines (Doubly so for people using Microsoft Defender, which takes up another spot). 

I think it is still a good idea to consider the impact of Apple's changes, however. It may be useful to make mention of it in the settings page, so that users looking to reactivate the icon are guided to the right place (Mac Settings > Menu Bar) when they are looking for it in the Docker settings. Perhaps it is possible to show a one-time warning to users that remove the icon from the menu bar, or add a section only when the menu bar icon is removed. That's my two cents, at least.

**Comment by brodyhoskins** [comment]
Is there some reason that Docker Desktop doesn't run as a `launchd` daemon and the UI just be a "normal" app that doesn't hog up space in the menu bar? It's incredibly frustrating being unable to open the UI because the icon is missing.

**Comment by aofei** [comment]
https://github.com/user-attachments/assets/e9227a37-a1d8-4cd2-abd4-d788064a7236

macOS 26 🎉

**Comment by brodyhoskins** [comment]
> https://github.com/user-attachments/assets/e9227a37-a1d8-4cd2-abd4-d788064a7236
> 
> macOS 26 🎉

Incredible! I'm not sure why there's a trend for macOS apps to follow the Windows "system tray" design pattern but I'm happy macOS allows cleaning up these gremlins.
